Dockerfile

```diff
@@ -0,0 +1,22 @@
+# Build Stage
+FROM rust:1.72.1 as build
+COPY /node/ /node/
+COPY Makefile .
+WORKDIR /node
+RUN apt-get update && apt-get install -y libclang-dev
+RUN cargo build --release
```

Comment:
Following the discussion you had about caching:
The image build runs line 6 and 7 on every code change, which takes very long time. 
I think it's possible to better use layer caching so that at least line 6 won't be repeated (I'm not sure why it does).

As for line 7, as you already mentioned, it should be possible to avoid the reinstalling of dependencies, by only copying `Cargo.toml` and `Cargo.lock` files, then create a dummy `main.rs` (`RUN echo "fn main() {println!(\"dummy\")}" > src/main.rs`) and do cargo build to cache the dependencies. Then, copy the rest of the source code and build again. 

This is not critical for now, but might worth trying.

Reply:
Cargo.toml is a workspace manifest, so it would need to be converted to a crate manifest.

Reply:
Exactly, that was my initial approach for caching external dependencies but since it works as a workspace manifest all the local code would be necessary to build along them.